snapshotter/devmapper/snapshotter.go

```diff
@@ -186,11 +206,46 @@ func (s *Snapshotter) Commit(ctx context.Context, name, key string, opts ...snap
 	log.G(ctx).WithFields(logrus.Fields{"name": name, "key": key}).Debug("commit")
 
 	return s.withTransaction(ctx, true, func(ctx context.Context) error {
-		_, err := storage.CommitActive(ctx, key, name, snapshots.Usage{}, opts...)
+		id, info, _, err := storage.GetInfo(ctx, key)
+		if err != nil {
+			return err
+		}
+
+		usage, err := s.getSnapshotUsage(ctx, id, &info)
+		if err != nil {
+			return err
+		}
+
+		_, err = storage.CommitActive(ctx, key, name, usage, opts...)
 		return err
 	})
 }
 
+// getSnapshotUsage returns data size consumed by snapshot layer itself (excluing parent data usage)
+func (s *Snapshotter) getSnapshotUsage(ctx context.Context, id string, info *snapshots.Info) (snapshots.Usage, error) {
+	deviceName := s.getDeviceName(id)
+	size, err := s.pool.GetUsage(deviceName)
+	if err != nil {
+		return snapshots.Usage{}, err
+	}
+
+	// Subtract parent data usage if any
+	if info.Parent != "" {
```

Comment:
Do you need to traverse through all of the ancestors?  It looks like this just accounts for the immediate parent.

Reply:
No, we need a difference between parent/child here. Have a look on `device1_snap` in PR description.

Reply:
Discussed offline, fixed in https://github.com/firecracker-microvm/firecracker-containerd/pull/147/commits/693f1f5d9a2fa7977cde92e314c999947673b0ae